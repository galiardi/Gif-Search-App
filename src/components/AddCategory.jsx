import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const trimedInputValue = inputValue.trim();
    if (trimedInputValue.length <= 1) return;
    onAddCategory(trimedInputValue);
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form' >
      <input type="text" placeholder="Busca gifs" value={inputValue} onChange={onInputChange} />
    </form>
  );
};

AddCategory.propTypes= {
  onAddCategory: PropTypes.func.isRequired,
}