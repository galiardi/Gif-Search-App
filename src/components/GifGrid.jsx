import PropTypes from 'prop-types'

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks';

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && <p>loading...</p>
      }
      <div className="card-grid">
        {gifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}