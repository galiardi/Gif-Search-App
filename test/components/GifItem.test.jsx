import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Testing GifItem', () => {

  const title = 'dirt collapse GIF';
  const url = 'https://media0.giphy.com/media/HlPP3bY8FRsWs/giphy.gif?cid=c04ecd26nljvmzpk5gxgmbxtkpapxjnhzy7192z7vzwiiasv&rid=giphy.gif&ct=g';


  test('debe hacer match con el snapshot', () => {

    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();

  });


  test('debe mostrar la imagen con el url y el alt indicado', () => {

    render(<GifItem title={title} url={url}/>);
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title)
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  })

  test('debe mostrar el titulo', () => {

    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy();
    
  });
});
