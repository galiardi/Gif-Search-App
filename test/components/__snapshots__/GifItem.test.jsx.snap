// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dirt collapse GIF"
      src="https://media0.giphy.com/media/HlPP3bY8FRsWs/giphy.gif?cid=c04ecd26nljvmzpk5gxgmbxtkpapxjnhzy7192z7vzwiiasv&rid=giphy.gif&ct=g"
    />
    <p>
      dirt collapse GIF
    </p>
  </div>
</div>
`;
